lib/transport/shell.js

```diff
@@ -109,7 +109,7 @@ Shell.prototype.transfer = function(files, remoteDir, options) {
                               (remote.username ? remote.username + '@' : ''),
                               remote.host, remoteDir);
 
-    var command = util.format('rsync --files-from %s %s --rsh="ssh -p%s%s" ./ %s',
+    var command = util.format('cat %s | xargs -n 50 -J % rsync %s --rsh="ssh -p%s%s" % %s',
```

Comment:
1. It's a chunk limit
1. See xargs doc http://charlesreid1.com/wiki/Xargs#-I_vs._-J
1. I hope :)
1. No, currently... But we can do some algorithm that will transform relative path into absolute. Question only one: what folder should be a "base"?
1. It will generate few rsync commands